connection.go

```diff
@@ -626,17 +542,18 @@ func (conn *Connection) dial() (err error) {
 	})
 
 	if err != nil {
+		c.Close()
 		return fmt.Errorf("unable to register watch: %w", err)
 	}
 
 	// Only if connected and fully initialized.
 	conn.lockShards()
-	conn.c = connection
+	conn.c = c
 	atomic.StoreUint32(&conn.state, connConnected)
 	conn.cond.Broadcast()
 	conn.unlockShards()
-	go conn.writer(w, connection)
-	go conn.reader(r, connection)
+	go conn.writer(c, c)
```

Comment:
Is there any motivation left to pass two arguments?

Reply:
It is necessary only from the responsibility side of the objects:

1. The first one is used to read/write data.
2. The second is needed to check: is the current connection still valid?

On the one hand, this can be a little confusing, on the other hand, it is clearer what object is used for.

Reply:
But I can merge it if you like. I don't think it's critical.

Reply:
Let's leave it as is for now